File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int max_length = 0;

        std::unordered_set<int> unique_nums(nums.begin(), nums.end());
        for (const int num: unique_nums) {
            if (unique_nums.count(num - 1)) {
                continue;
            }

            int cur_length = 0;
            int cur_num = num;
            while (unique_nums.count(cur_num)) {
                cur_length++;
                cur_num++;
            }

            max_length = std::max(max_length, cur_length);
        }

        return max_length;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        max_length = 0
        unique_nums = set(nums)

        for num in unique_nums:
            if num - 1 in unique_nums:
                continue
            
            cur_length = 0
            while num in unique_nums:
                num += 1
                cur_length += 1
            
            max_length = max(max_length, cur_length)

        return max_length